drivers/sensor/stm32_vbat/stm32_vbat.c

```diff
@@ -72,7 +71,8 @@ static int stm32_vbat_channel_get(const struct device *dev, enum sensor_channel
 		return -ENOTSUP;
 	}
 
-	voltage = data->raw * cfg->vref_mv / 0x0FFF; /* Sensor value in millivolts */
+	/* Sensor value in millivolts */
+	voltage = data->raw * adc_ref_internal(data->adc) / 0x0FFF;
 	/* considering the vbat input through a resistor bridge */
 	voltage = voltage * cfg->ratio / 1000; /* value of SENSOR_CHAN_VOLTAGE in Volt */
```

Comment:
note: this driver could likely be improved by using adc_dt_spec facilities (out of the scope of this PR)

Reply:
Yes, unfortunately this won't work in that case (or rather not without warnings in the DT generation) due to issue #47119.

That said, I have used that for the more generic implementation in PR #47176. Feedback is welcome.